### PR TITLE
Humble Puma Motor Driver

### PR DESCRIPTION
**Pull request by luis-camero**
Includes `clearpath_socketcan_interface` modified from `ros_canopen`. 

Should also rename main branch to `humble-devel`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,3 +2,5 @@ puma_motor_driver [![Build Status](https://travis-ci.org/clearpathrobotics/puma_
 =================
 
 A socketcan-based ROS driver for the Puma, a brushed DC motor driver based on the BDC-RDK24 kit from TI.
+
+Currently need to install the socket_can branch from [this pull request](https://github.com/ros-industrial/ros_canopen/pull/364).
```
Comment (tonybaltovski) [inline README.md]: Let's remove this comment.

File: clearpath_socketcan_interface/package.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format2.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="2">
+  <name>clearpath_socketcan_interface</name>
+  <version>0.8.2</version>
```
Comment (tonybaltovski) [inline clearpath_socketcan_interface/package.xml]: This needs to be changed to match the rest of the packages.

File: puma_motor_driver/CMakeLists.txt [hunk 3]
```diff
@@ -1,77 +1,98 @@
...
+set(Boost_USE_STATIC_LIBS OFF)
+set(Boost_USE_MULTITHREADED ON)
+set(Boost_USE_STATIC_RUNTIME OFF)
+find_package(Boost REQUIRED)
+####################################################
 
 include_directories(
   include
-  ${catkin_INCLUDE_DIRS}
+	${Boost_INCLUDE_DIRS}
```
Comment (tonybaltovski) [inline puma_motor_driver/CMakeLists.txt]: Spacing

File: puma_motor_driver/package.xml [hunk 4]
```diff
@@ -7,17 +7,18 @@
...
   <depend>sensor_msgs</depend>
-  <depend>serial</depend>
+  <!-- <depend>serial_driver</depend> -->
 
   <exec_depend>can-utils</exec_depend>
   <exec_depend>joy</exec_depend>
 
-  <test_depend>roslaunch</test_depend>
-  <test_depend>roslint</test_depend>
+	<export>
```
Comment (tonybaltovski) [inline puma_motor_driver/package.xml]: Spacing

File: puma_motor_driver/src/driver.cpp [hunk 5]
```diff
@@ -199,90 +201,93 @@ void Driver::commandDutyCycle(const float cmd)
 void Driver::commandSpeed(const double cmd)
 {
   // Converting from rad/s to RPM through the gearbox.
-  sendFixed16x16((LM_API_SPD_SET | device_number_), (cmd * radPerSecToRpm()));
+  double rpm = cmd * radPerSecToRpm();
```
Comment (tonybaltovski) [inline puma_motor_driver/src/driver.cpp]: Do went want to revert this?

File: puma_motor_driver/src/multi_puma_node.cpp [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+    received_fields |= driver.receivedBusVoltage() << StatusBit::BusVoltage;
+    received_fields |= driver.receivedOutVoltage() << StatusBit::OutVoltage;
+    received_fields |= driver.receivedAnalogInput() << StatusBit::AnalogInput;
+    received_fields |= 1 << StatusBit::AnalogInput;
+    received_fields |= driver.receivedTemperature() << StatusBit::Temperature;
+    received_fields |= driver.receivedMode() << StatusBit::Mode;
+    received_fields |= driver.receivedFault() << StatusBit::Fault;
+    if (received_fields != (1 << StatusBit::Count) - 1)
+    {
+      //RCLCPP_INFO(this->get_logger(), "Received Status Fields %x", received_fields);
```
Comment (tonybaltovski) [inline puma_motor_driver/src/multi_puma_node.cpp]: Should we enable these are DEBUG?

File: clearpath_socketcan_interface/package.xml (hunk 2 above)
Comment (luis-camero) [reply clearpath_socketcan_interface/package.xml]: I changed it but what about the changelog file?

Comment (tonybaltovski) [reply clearpath_socketcan_interface/package.xml]: Just just get rid of the changelog.

**Review by tonybaltovski — APPROVED** [review]

**Pull request merged**